src/pypromice/postprocess/get_bufr.py

```diff
@@ -202,9 +202,6 @@ def get_bufr(
             logger.info("Current data is not newer than latest")
             continue
 
-        if stid not in latest_timestamps:
```

Comment:
As I recall, `latest_timestamps.pickle` was completely "self-maintaining".

In existing version, with each `stid` that is processed we set:

`current_timestamps[stid] = current_timestamp`

...before checking `if stid in latest_timestamps:`.

Then, if this is a new station, it will be written back to the pickle file with:

```
	# Write the most recent timestamps back to the pickle on disk
	print('writing latest_timestamps.pickle')
	with open(args.timestamps_pickle_filepath, 'wb') as handle:
		pickle.dump(current_timestamps, handle, protocol=pickle.HIGHEST_PROTOCOL)
```
I would need to check, but perhaps the very first time this station is processed, it would be skipped until it is in the file?

The important part was to have a `current_timestamp` and a `latest_timestamp`, so that each processing run could compare:

`if (current_timestamp > latest_timestamp) and (current_timestamp > two_days_ago):`

This is to make sure we are submitting new data, and recent data. The pickle file was simply a way to persist the last processed timestamps between runs.

In any case, the idea was to never have to touch the pickle file. If a new station is added to the wmo config file, it should "just work".